maas/resource_maas_network_interface_tag_test.go

```diff
@@ -0,0 +1,138 @@
+package maas_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"terraform-provider-maas/maas"
+	"terraform-provider-maas/maas/testutils"
+)
+
+func TestSplitTagStateId(t *testing.T) {
+	expectedSystemId := "acb123"
+	expectedInterfaceId := 12
+	stateId := fmt.Sprintf("%s:%d", expectedSystemId, expectedInterfaceId)
+	systemId, interfaceId, err := maas.SplitTagStateId(stateId)
+	if err != nil {
+		t.Fatalf("Error splitting state ID: %s", err)
+	}
+	if systemId != expectedSystemId || interfaceId != expectedInterfaceId {
+		t.Fatalf("Expected system ID %s and interface ID %d, got system ID %s and interface ID %d", expectedSystemId, expectedInterfaceId, systemId, interfaceId)
+	}
+}
+
+func TestAccNetworkInterfaceTag_basic(t *testing.T) {
+	hostname := acctest.RandomWithPrefix("tf")
+	macAddress := testutils.RandomMAC()
+	tagName := acctest.RandomWithPrefix("tag")
+	tagName2 := acctest.RandomWithPrefix("tag2")
```

Comment:
tag2 can also be tag since it is a prefix. Let's remove 2 and 3 from the next line